x/programs/rust/wasmlanche_sdk/src/state.rs

```diff
@@ -77,3 +78,40 @@ impl State {
         from_slice(&val).map_err(|_| StateError::InvalidBytes)
     }
 }
+
+/// Converts a raw pointer to a deserialized value.
+/// Expects the first 4 bytes of the pointer to represent the [length] of the serialized value,
+/// with the subsequent [length] bytes comprising the serialized data.
+// pub fn from_raw_ptr<V>(ptr: i64) -> V
+// where
+//     V: serde::de::DeserializeOwned,
+// {
+//     let (bytes, _) = bytes_and_length(ptr);
+//     from_slice(&bytes).expect("failed to deserialize")
+// }
+
+// Converts a raw pointer to a type that implements the Argument trait.
+// Expects the first 4 bytes of the pointer to represent the [length] of the serialized value,
+// with the subsequent [length] bytes comprising the serialized data.
+#[must_use]
+pub fn from_raw_ptr<V>(ptr: i64) -> V
+where
+    V: Argument,
+{
+    let (bytes, _) = bytes_and_length(ptr);
+    V::from_bytes(&bytes)
+}
+
+// TODO: move this logic to return a Memory struct that conatins ptr + length
+/// Returns a tuple of the bytes and length of the argument.
+/// # Panics
+/// Panics if the value cannot be converted from i32 to usize.
+/// # Safety
+/// This function is unsafe because it dereferences raw pointers.
+#[must_use]
+pub fn bytes_and_length(ptr: i64) -> (Vec<u8>, i32) {
```

Comment:
Why return an `i32` for the length here?

Reply:
yep should be a u32